iModelCore/ECDb/Tests/NonPublished/CommonTableExpTests.cpp

```diff
@@ -1594,14 +1594,139 @@ TEST_F(CommonTableExpTestFixture, CTE_Subquery_Tests) {
+        ASSERT_STREQ(stmt.GetNativeSql(),"SELECT [ECClassDef].[ECInstanceId],[ECClassDef].[ECClassId],[ECClassDef].[SchemaId],[ECClassDef].[SchemaRelECClassId],[ECClassDef].[Name],[ECClassDef].[DisplayLabel],[ECClassDef].[Description],[ECClassDef].[Type],[ECClassDef].[Modifier],[ECClassDef].[CustomAttributeContainerType],[ECClassDef].[RelationshipStrength],[ECClassDef].[RelationshipStrengthDirection] FROM (SELECT [Id] ECInstanceId,37 ECClassId,[SchemaId],(CASE WHEN [SchemaId] IS NULL THEN NULL ELSE 38 END) [SchemaRelECClassId],[Name],[DisplayLabel],[Description],[Type],[Modifier],[CustomAttributeContainerType],[RelationshipStrength],[RelationshipStrengthDirection] FROM [main].[ec_Class]) [ECClassDef] WHERE NOT EXISTS(WITH a AS (SELECT [ECPropertyDef].[ECClassId] [K0] FROM (SELECT [Id] ECInstanceId,44 ECClassId FROM [main].[ec_Property]) [ECPropertyDef])\nSELECT [K0] FROM a WHERE [K0] >= [ECClassDef].[ECInstanceId])" );
+    }
+    if ("simple_select_cte_without_columns_subquery_ANY") {
+        auto ecsql = R"(select * from meta.ECClassDef where ECInstanceId >= ANY(with a AS (select * from meta.ClassHasBaseClasses) select * from a))";
+        ECSqlStatement stmt;
+        ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, ecsql));
+        ASSERT_STREQ(stmt.GetNativeSql(),"SELECT [ECClassDef].[ECInstanceId],[ECClassDef].[ECClassId],[ECClassDef].[SchemaId],[ECClassDef].[SchemaRelECClassId],[ECClassDef].[Name],[ECClassDef].[DisplayLabel],[ECClassDef].[Description],[ECClassDef].[Type],[ECClassDef].[Modifier],[ECClassDef].[CustomAttributeContainerType],[ECClassDef].[RelationshipStrength],[ECClassDef].[RelationshipStrengthDirection] FROM (SELECT [Id] ECInstanceId,37 ECClassId,[SchemaId],(CASE WHEN [SchemaId] IS NULL THEN NULL ELSE 38 END) [SchemaRelECClassId],[Name],[DisplayLabel],[Description],[Type],[Modifier],[CustomAttributeContainerType],[RelationshipStrength],[RelationshipStrengthDirection] FROM [main].[ec_Class]) [ECClassDef] WHERE EXISTS(WITH a AS (SELECT [ClassHasBaseClasses].[ECInstanceId] [K0],[ClassHasBaseClasses].[ECClassId] [K1],[ClassHasBaseClasses].[Ordinal] [K2],[ClassHasBaseClasses].[SourceECInstanceId] [K3],[ClassHasBaseClasses].[SourceECClassId] [K4],[ClassHasBaseClasses].[TargetECInstanceId] [K5],[ClassHasBaseClasses].[TargetECClassId] [K6] FROM (SELECT [ec_ClassHasBaseClasses].[Id] [ECInstanceId],42 [ECClassId],[ec_ClassHasBaseClasses].[ClassId] [SourceECInstanceId],37 [SourceECClassId],[ec_ClassHasBaseClasses].[BaseClassId] [TargetECInstanceId],37 [TargetECClassId],[Ordinal] FROM [main].[ec_ClassHasBaseClasses]) [ClassHasBaseClasses])\nSELECT [K0],[K1],[K2],[K3],[K4],[K5],[K6] FROM a WHERE [ECClassDef].[ECInstanceId] >= [K0] OR [ECClassDef].[ECInstanceId] >= [K1] OR [ECClassDef].[ECInstanceId] >= [K2] OR [ECClassDef].[ECInstanceId] >= [K3] OR [ECClassDef].[ECInstanceId] >= [K4] OR [ECClassDef].[ECInstanceId] >= [K5] OR [ECClassDef].[ECInstanceId] >= [K6])" );

```

Comment:
Hey, I while the changes themselves look fine to me, these tests are not good. I see you already pushed similar tests to this file in a previous "Added support for CTE in subquery PR#836".
I know some of our tests check for string equality against the NativeSQL, but it's a bad habit. Just look at these tests. It's near impossible to tell what is wrong if one of them starts failing. The NativeSql depends on so many factors and a slight change in a future story may cause it to look different. Then, somebody has to go and adjust all these expected results by hand.

It's much better to test against the actual results and column infos of the query. It looks to me as if that's possible for these queries. These enormous NativeSql strings are going to be a pain to deal with in the future if we don't adjust it now.

Reply:
Sure I will be adding tests against real data as well then tomorrow